README.md

```diff
@@ -1,112 +1,141 @@
+`npm install @acpaas-ui-widgets/ngx-location-picker`
 
-Import the component in your module:
+Then import the component inside your module:
 
 ```ts
-import { LocationPickerModule } from '@acpaas-ui-widgets/ngx-location-picker';
+import {NgxLocationPickerModule} from '@acpaas-ui-widgets/ngx-location-picker';
 
 @NgModule({
   imports: [
     ...,
-    LocationPickerModule
+    NgxLocationPickerModule
   ],
   ...
 })
 ```
 
-In the index.html, include the core branding stylesheet:
+Finally include the required styles:
+
+Add Antwerp core branding stylesheet in your index.html file:
 
 ```html
 <link rel="stylesheet" href="https://cdn.antwerpen.be/core_branding_scss/3.0.3/main.min.css">
 ```
 
-> For projects that are still using Angular 5, we are [maintaining a v1 branch](https://github.com/digipolisantwerp/location-picker_widget_angular/tree/v1), which will still receive bug fixes if needed.
+Add required leaflet styles in the styles array in your angular.json project file.
 
-```sh
-> npm install @acpaas-ui-widgets/ngx-location-picker@"<2.0.0"
+```
+"node_modules/leaflet/dist/leaflet.css"
+"node_modules/leaflet-draw/dist/leaflet.draw.css"
 ```
 
-### In your template:
+##### Getting an API key
 
-```html
-<aui-location-picker
-    url="http://localhost:9999/api"
-    [(value)]="location">
-</aui-location-picker>
-```
+This component uses a dedicated API for which an api key is required. Request a key through one of the links below (depending on you environment):
 
-(replace the url of the BFF service)
+Development: https://api-store-o.antwerpen.be  
+Acceptance: https://api-store-a.antwerpen.be  
+Production: https://api-store.antwerpen.be  
 
-In the component code:
+Create a contract with the Location Picker api for your organization.
 
-```ts
-class YourComponent {
+##### Usage
 
-    // you can assign an initial value here
-    location: LocationPickerValue;
+```html
+<ngx-location-picker
+    [baseUrl]="baseUrl"
+    [defaultZoom]="defaultZoom"
+    [onSelectZoom]="onSelectZoom"
+    [mapCenter]="mapCenter"
+    [hasSidebar]="hasSidebar"
+    [showMap]="showMap"
+    [featureLayers]="featureLayers"
+    [placeholder]="placeholder"
+    [label]="label"
+    [noResultsLabel]="noResultsLabel"
+    (addPolygon)="onAddPolygon($event)"
+    (addLine)="onAddLine($event)"
+    (editFeature)="onEditFeature($event)"
+    (locationSelect)="onLocationSelect($event)"
+></ngx-location-picker>
+```
 
-    ...
+```ts
+class ExampleComponent {
+    
+    /**
+     * Url to the backend-for-frontend (bff) Should function as pass through to the Location Picker API.
+     * Make sure your BFF adds the api key you requested earlier.
+     */
+    baseUrl: string = 'https://path-to-bff';
+    /* the default zoom level on map load. */
+    defaultZoom: number = 14;
+    /* the zoom level when a location is selected. */
+    onSelectZoom: number = 16;
+    /* the initial map center on load. */
+    mapCenter: Array<number> = [51.215, 4.425];
```

Comment:
does this replace the location property for setting an initial location? what if the application just has an address and wants to show this field prefilled? how do you use this in a form which was already filled out and address is one of the filled out values?

Reply:
Initial location can now be set using [(NgModel)] or a reactive form control